elide-async/src/main/java/com/yahoo/elide/async/hooks/AsyncAPIHook.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.async.hooks;
+
+import static com.yahoo.elide.annotation.LifeCycleHookBinding.Operation.CREATE;
+import static com.yahoo.elide.annotation.LifeCycleHookBinding.Operation.READ;
+import static com.yahoo.elide.annotation.LifeCycleHookBinding.TransactionPhase.POSTCOMMIT;
+import static com.yahoo.elide.annotation.LifeCycleHookBinding.TransactionPhase.PRESECURITY;
+
+import com.yahoo.elide.annotation.LifeCycleHookBinding;
+import com.yahoo.elide.async.models.AsyncAPI;
+import com.yahoo.elide.async.models.AsyncAPIResult;
+import com.yahoo.elide.async.models.QueryStatus;
+import com.yahoo.elide.async.service.AsyncAPIThread;
+import com.yahoo.elide.async.service.AsyncExecutorService;
+import com.yahoo.elide.core.exceptions.InvalidOperationException;
+import com.yahoo.elide.security.RequestScope;
+
+import lombok.Data;
+
+import java.security.Principal;
+import java.util.concurrent.Callable;
+
+/**
+ * AsyncAPI Base Hook methods.
+ */
+@Data
+public abstract class AsyncAPIHook {
```

Comment:
I think this should implement LifeCycleHook to create a "Is A" relationship here.